src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -167,7 +167,7 @@ private static void AddAllocatedEndpointInfo(IEnumerable<AppResource> resources)
                     sp.ServiceBindingAnnotation.Name,
                     PortProtocol.ToProtocolType(svc.Spec.Protocol),
                     svc.AllocatedAddress!,
-                    (int)svc.AllocatedPort!,
+                    (int)(sp.ServiceBindingAnnotation.Port ?? svc.AllocatedPort!), // Checked by HasCompleteAddress above.
```

Comment:
This works because if you have just one replica, you can hit the replica directly and it is pretty much the same thing as hitting the proxy for the service that uses a single replica. But this breaks if you start from one replica and then want to scale up to multiple replicas. Granted, we do not have this scenario today in DAB model.

This change also introduces a subtle mismatch between DAB and DCP model: if you use `portFor` environment variable function, it will tell you a different port for a service from what DAB thinks is the port for the same service is (again, for a single-replica service).

But all this said, I think this is fine as an intermediate step.

Reply:
The intention behind `AllocatedEndpointAnnotation.Port` is that it is a way for DCP to communicate back to the app model what the front-end port has been allocated to a service. If a service allocates a port via `ServiceBindingAnnotation.Port` then `AllocatedEndpointAnnotation.Port` should always be the same. Where `ServiceBindingAnnotation.Port` is null, that is a signal for DCP to allocate a random port, and that random port should be set in `AllocatedEndpointAnnotation.Port` so that things like environment callbacks can get the actual port to set connection strings etc.

Reply:
... I'm not sure how portFor comes into this? I would have thought that at the point that we are constructing the `AllocatedEndpointAnnotation` that we would have this information all setup.